packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -200,7 +211,11 @@ export class RouterliciousRestWrapper extends RestWrapper {
 				},
 			};
 		}
+
 		// Failure
+		// on failure, add the request entry into the retryCounter map to count the subsequent retries
+		this.retryCounter.set(requestKey, requestRetryCount ? requestRetryCount + 1 : 1);
```

Comment:
You need to do this in the catch of when you make the request. As that would just throw and not reach here.